source/Mission.cpp

```diff
@@ -834,7 +837,9 @@ string Mission::BlockedMessage(const PlayerInfo &player)
 	if(cargoNeeded < 0 && bunksNeeded < 0)
 		return "";
 	
-	map<string, string> subs;
+	map<string, string> subs = substitutions.Substitutions(player);
+	auto globalSubs = GameData::GetTextReplacements().Substitutions(player);
```

Comment:
Isn't it more efficient to load the globalSubs first and then append the mission substitutions? If we change the code of the Substitutions() call to get the map as parameter, then they can just be added to the end directly.

Reply:
It's done in this order because when inserting into a map, if a key already exists then the new value is ignored. So since we want mission substitutions to override global substitutions, mission substitutions are loaded first and global substitutions are inserted.

Reply:
I was thinking of changing the signature of Substitutions in TextReplacements to:
`void Substitutions(std::map<std::string, std::string> subs, const std::map<std::string, int64_t> &conditions) const;`
This way it is possible to create the map inside Mission.cpp (as is already done now), then add the global substitutions and then add the mission-specific substitutions.

Reply:
Okay, I see what you mean.

Reply:
Is the latest commit what you had in mind?

Reply:
Yes that was exactly what I had in mind, and sorry for the bug in my example where subs wasn't a reference.